src/test/java/bio/terra/stairway/ScenarioTest.java

```diff
@@ -131,6 +135,209 @@ public void testUndo() throws Exception {
         assertTrue(file.exists());
     }
 
+    @Test
+    public void testQuietDown() throws Exception {
+        String inResult = "quieted down and woke up";
+        FlightMap inputParameters = new FlightMap();
+        inputParameters.put(MapKey.CONTROLLER_VALUE, 1);
+        inputParameters.put(MapKey.RESULT, inResult);
+
+        TestStopController.setControl(0);
+        String flightId = stairway.createFlightId();
+
+        stairway.submit(flightId, TestFlightQuietDown.class, inputParameters);
+        // Allow time for the flight thread to go to sleep
+        TimeUnit.SECONDS.sleep(5);
+
+        // Quiet down - don't wait long; we want control so we can unblock the thread!
+        boolean quietYet = stairway.quietDown(1, TimeUnit.SECONDS);
+        assertFalse(quietYet, "Not quiet yet");
+
+        // Wake up the thread; it should exit into READY state
+        TestStopController.setControl(1);
+        // Allow time for the flight thread to wake up and exit
+        TimeUnit.SECONDS.sleep(5);
+
+        FlightState state = stairway.getFlightState(flightId);
+        assertThat("State is ready", state.getFlightStatus(), equalTo(FlightStatus.READY));
+        assertNull(state.getStairwayId(), "Flight is unowned");
```

Comment:
I'm confused about the expected order of events here. At this point, the sleeping flight thread either woke up in time to finish the step inside of the thread pool's termination grace period, or it didn't wake up in time and got an interrupt. In both cases, I think you'd get a STEP_RESULT_STOP, which would turn into a FlightStatus.READY, so that check makes sense. But the current step would be different because in one case it was completed successfully and in one case it was interrupted. Am I following this correctly?

Regardless, I think it would be good to test both cases -- interrupt because step did/not finish within quiet down grace period.

Reply:
In this test case, there is never an interrupt. The only way for the flight to get to the ready state is if it correctly detects the QuietDown state and stops itself.

There is a test for the terminate case: `testTerminate()` later in the module.

Reply:
By grace period, I meant the timeout in quietDown->awaitTermination -- I think that throws an interrupt after the timeout expires. How do you know that the sleep step exits before that quietdown timeout?

Yes, I do see the terminate test, looks great.